### DISCUSSION TITLE
Can't install for multiple users

### DISCUSSION THREAD

**Discussion by ai-niki** [Homebrew/discussions#1762] [Getting started] [answered]
Hi!

I'm trying to install brew for 2 admin users on latest Big Sur. I've tried almost every approach from [stackoverflow](https://stackoverflow.com/questions/41840479/how-to-use-homebrew-on-a-multi-user-macos-sierra-setup) but they all come down to these simple commands:
```
chgrp -R admin /usr/local/*
chmod -R g+w /usr/local/*
```
This does the job for the user which executes the commands. While the other user always gets these errors on brew update:
```
...
error: unable to unlink old 'Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-rails-2.10.1/lib/rubocop/rails/inject.rb': Permission denied
error: unable to unlink old 'Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-rails-2.10.1/lib/rubocop/rails/schema_loader.rb': Permission denied
error: unable to unlink old 'Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-rails-2.10.1/lib/rubocop/rails/schema_loader/schema.rb': Permission denied
error: unable to unlink old 'Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-rails-2.10.1/lib/rubocop/rails/version.rb': Permission denied
```

What is the correct way to install Home-brew for 2 admin users?

**Comment by gromgit** [comment, marked as answer]
Create a third admin user (e.g. `homebrew`), then:
```
sudo chown -R homebrew:admin /usr/local/*
```
Then all further _write_ operations (i.e. `install`, `update`, `upgrade`, etc.) should be done by everyone else as that user, by the power of `su` or `sudo`, depending on your preference:
```
su - homebrew brew update
```
In this respect, there's only one difference between Homebrew and a system package manager: the former lets you decide which user to grant ownerships to, while the latter fixes it as `root`.

**Reply by ai-niki** [reply]
In that case it's just enough to run `su user1 -c 'brew update'`, innit? No need in creating a separate user.

**Reply by gromgit** [reply]
My personal preference is to have a mission-specific account, but if `user1` doesn't mind opening up her account, then as long as `user1` owns the Homebrew hierarchy, that could work too.

**Reply by ai-niki** [reply]
Thanks! It works!

**Reply by dijonkitchen** [reply]
I wasn't able to do `su - some-user brew update` on a new Mac with two users. 

You'll need the `-c 'some command'` for it to work. I also wanted to note that only `su` works, not `sudo`. 

TLDR: `su - some-user -c 'some command'`

Edit: 
I was able to use `sudo` when adding some more flags: `sudo -Hiu some-user some-long-command`
